modcc/functioninliner.cpp

```diff
@@ -76,9 +59,78 @@ expression_ptr inline_function_call(Expression* e)
                                      e->location());
         }
 
-        return new_e;
+        if (e->is_assignment()) {
+            e->is_assignment()->replace_lhs(lhs->clone());
+        }
+    };
+
+    if (exp->is_if()) {
+        if (!exp->is_if()->false_branch()) {
+            throw compiler_exception(
+                    "can only inline if statements with associated else", exp->location()
+            );
+        }
+        // Modify the condition
```

Comment:
What does this comment mean?

Reply:
It means we are inlining the condition of the if condition (replacing the function arguments with the call arguments)
I can fix to : Modify the condition of the if statement / Substitute arguments in if condition ?